### PR TITLE
Integrate Facebook Ads into Android Application in Android Studio

### LINKED ISSUE #4585: [API] Integrating Facebook Ads with your Android Application in Android studio
#### Introduction
Besides developing mobile applications for technological adjustments, developers always look towards getting something in return. One of the greatest ways of earning from applications being advertisement, this article shows developers how they can integrate banner and interstitial ads in their application.

With Facebook Audience Network, developers are allowed to monetize their android applications with Facebook Ads. This motivates developers to come up with new applications every now and then.

#### Key takeaways 
- Overview of all the requirements
- Set up your Android application project
- Installation of the required **sdk** i.e **(Facebook Network Audience) sdk** 
       and other requirements.
- Steps of **Integrating**FAN sdk and adapter into your Android application

#### Article quality
The official documentation gives a general directive using the initial generated code, which sometimes proves hard to follow.
I intend to give a clear step by step explanation with more understanding simplicity and flexibility with different versions (not offered in the available documentation).

#### References
N/A

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by del-ui**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #4585 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by del-ui** [comment]
@Jethro-magaji I have done some refinement on this article to enhance readability as directed earlier

**Comment by Jethro-magaji** [comment]
> @Jethro-magaji I have done some refinement on this article to enhance readability as directed earlier

Please take your time to make serious edits on your article and don't rush to submit. Because your PR was closed twice due to quality and this is the third time i will be making a review on your article, which takes time from your reviewer. 
I highly recommend you get a good editor to edit your article before submitting. However, if you're sure that the current PR is good to go, then i can proceed with the review but note that it would be closed if all doesn't go well.
Thanks.

Cc @WanjaMIKE  @hectorkambow

**Comment by ahmadmardeni1** [comment]
This is the last round of revision @Jethro-magaji If the article is not ready then please close it and the author can submit a new topic after going through our docs.
Our team does NOT provide editing services. @del-ui Thank you.

File: content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+- [Facebook Audience Network SDK](#https://developers.facebook.com/resources/audience-network-sdk-6.7.0.zip)
+- [Facebook Audience Network account](#https://business.facebook.com/pub/home/?source=help&business_id=1101812683558213&global_scope_id=1101812683558213)
+- [Android Studio](#https://developer.android.com/studio/releases)
+- [Facebook business account](#https://business.facebook.com/overview)
+
+You also need to be familiar with the layout and how to use Android studio. With all these requirements in place, you can move on to the steps involved in the integration process.
+Assuming you have your Android studio up and running, the only thing you have to install is the FAN SDK.
+
+**FAN (Facebook Audience Network) SDK installation**
+To install, you can download your SDK manually from [here](https://developers.facebook.com/resources/audience-network-sdk-6.8.0.zip) (Not recommended) or follow the steps in [Initial integration steps](#initial-integration-steps) to install it the right way.
```
Comment (Jethro-magaji) [inline content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md:28]: Rephrase this sentence, emphasis on recommended method.

File: content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+### Initial integration steps
+Before beginning the integration process, you have to first set up your Facebook Business Manager account for the ads.
+The steps below will guide you in setting up your account.
+
+#### Step 1. **Set-up a  Business Page**
+The first and foremost step is to set up a business page where your ads will appear. You can follow this short [video](https://www.youtube.com/watch?v=fzW4eHQQLFk) to see how you can create your page.
```
Comment (Jethro-magaji) [inline content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md:35]: No need for "First & foremost".

File: content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+#### Step 2 **Set up a Facebook Manager  account**
+To create a Facebook manager account, go to https://business.facebook.com/overview to first create a Facebook manager account as shown in the picture below.
+![create](engineering-education/integrating-facebook-ads-with-your-android-application-in-android-studio
+/create.png)
+
+#### Step 3 **Open a Facebook Audience Network (FAN) account**
+Set up a FAN account using the business manager account you created to make sure they are all integrated.
+
+#### Step 4 Build Application properties for your account
+Once you have created your FAN account, click on the extension/dropdown button at the top and select the monetization manager. Click on integration>properties, key in your Application name and id then select your platform, either Android or iOS.
```
Comment (Jethro-magaji) [inline content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md:46]: Edit this "Click on integration>".

File: content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+If you don't have a ready App, follow these [directives](#integration-in-android-studio) to create a simple test application.
+
+#### Step 5 Select the type of Ads
+The final initial step is to select and create the ad type.
+Once done, you are ready for the integration process in the Android studio.
+
+
+### Integration in Android Studio
+Don't have an Application? No need to worry since you can set up a simple application in a few steps below.
```
Comment (Jethro-magaji) [inline content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md:56]: Rephrase this sentence.

File: content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+implementation 'com.facebook.android:audience-network-sdk:5.9.1'
+
+```
+This dependency integrates the SDK into your Application.
+
+![sdk](engineering-education/integrating-facebook-ads-with-your-android-application-in-android-studio
+/sdk.png)
+
+#### Adding the Ads to your App
+In this case, we are going to add Banner Ads as follows;
```
Comment (Jethro-magaji) [inline content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md:75]: Use second person "You" not "We".

File: content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+Consumers prefer in-App advertising over display advertising on mobile websites. Smartphone consumers find ads on the mobile web to be compacted and unpleasant. In-App advertisements are resized to fit the screen and seem better, increasing overall engagement.
+
+When establishing in-App advertisements, **accessibility** is another important issue to consider. In-App mobile advertising is easier to reach because most consumers have their phones with them all the time, wherever they go. You have access to a captive audience or users who are already using the App and are interested in what you have to say. It's easier to convert them because they're already interested.
+
+### Demerits of integrating into some Applications
+In-App advertisement as seen earlier has many advantages. Besides, it is not always Appealing to some users when ads keep popping in all the time.
+
+Here are some demerits in an in-App advertisement;
+
+- People/users might hate the App due to **disturbances from ads pop-up**. This may discourage users from using known Applications with in-App advertisements.
```
Comment (Jethro-magaji) [inline content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md:144]: The bold sentences are supposed to be at the beginning of the points as headings and not inline sentences.

File: content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+This dependency integrates the SDK into your Application.
+
+![sdk](engineering-education/integrating-facebook-ads-with-your-android-application-in-android-studio
+/sdk.png)
+
+#### Adding the Ads to your App
+In this case, we are going to add Banner Ads as follows;
+
+On the XML file(activity.xml), add a layout for the Banner Ads and implement it in the java file.This code below does everything for you.
+```kotlin
```
Comment (Jethro-magaji) [inline content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md:78]: Add comments of what the code does or below the code.

**Review by Jethro-magaji — CHANGES_REQUESTED** [review]
Hi @del-ui 
I still think your article has more work to be done, I suggest running your article on tools like Grammarly for corrections. I have dropped some reviews please go over them and also check the section EngEd docs.
Thanks.

**Comment by del-ui** [comment]
@Jethro-magaji I have everything in place

File: content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md [hunk 8]
```diff
@@ -1,19 +1,5 @@
...
-excerpt_separator: <!--more-->
-images:
-
-  - url: /engineering-education/facebook-ads-in-android-studio/hero.png
-    alt: Facebook Ads Android App in Android Studio
----
+### Integrating Facebook Ads into your Android App in Android Studio
+
 ### Introduction
 Monetizing an application is just as an important goal for a developer as producing the required product. This motivates developers and programmers to come up with more software with a financial thought in mind which triggers quality.
```
Comment (Jethro-magaji) [inline content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md:4]: Please rephrase this "Monetizing an application is just as an important goal for a developer as producing the required product. This motivates developers and programmers to come up with more software with a financial thought in mind which triggers quality."

File: content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+/output.png)
+
+You can alter the code to and other types of Ads like interstitial Ads and Native Ads in the same manner.
+
+### Why integrate Facebook ads?
+Consumers prefer in-App advertising over display advertising on mobile websites. Smartphone consumers find ads on the mobile web to be compacted and unpleasant. In-App advertisements are resized to fit the screen and seem better, increasing overall engagement.
+
+When establishing in-App advertisements, **accessibility** is another important issue to consider. In-App mobile advertising is easier to reach because most consumers have their phones with them all the time, wherever they go. You have access to a captive audience or users who are already using the App and are interested in what you have to say. It's easier to convert them because they're already interested.
+
+### Demerits of integrating into some Applications
```
Comment (Jethro-magaji) [inline content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md:139]: Integrating what? In-App ads or what?

File: content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+In-App advertisement as seen earlier has many advantages. Besides, it is not always Appealing to some users when ads keep popping in all the time.
+
+Here are some demerits in an in-App advertisement;
+
+- People/users might hate the App due to **disturbances from ads pop-up**. This may discourage users from using known Applications with in-App advertisements.
+
+- For some people, it may be **difficult to navigate**. Some interested customers may not get to see the advertisement due to the incompatibility of the ads with some phones.
```
Comment (Jethro-magaji) [inline content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md:146]: This "**difficult to navigate**"should be at the beginning of the sentence

File: content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+
+- **Cost incurred on the user** to get the advertisement. To get the advertisement, the user's internet bundles get consumed from an unintended advertisement viewing task.
+
+### Conclusion
+At this point, you should be able to integrate your application with Facebook Ads for added monetary advantage. However, this might not be an option for many developers depending on the kind of application they are developing. 
+The verdict on whether to integrate your application with these Ads is always the owners' preference.
```
Comment (Jethro-magaji) [inline content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md:152]: The owner in what context are you taking about here. Please rephrase this sentence.

File: content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md [hunk 12]
```diff
@@ -22,10 +22,10 @@ The key requirements for a successful integration include:
 - [Facebook business account](https://business.facebook.com/overview)
 
 You also need to be familiar with the layout and how to use Android studio. With all these requirements in place, you can move on to the steps involved in the integration process.
-Assuming you have your Android studio up and running, the only thing you have to install is the FAN SDK.
+With your Android studio installed and working properly, you will needto install a Facebook Audience Network (FAN) account sdk.
```
Comment (Jethro-magaji) [inline content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md:25]: "needto" give some spacing in between

File: content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md [hunk 13]
```diff
@@ -22,10 +22,10 @@ The key requirements for a successful integration include:
 - [Facebook business account](https://business.facebook.com/overview)
 
 You also need to be familiar with the layout and how to use Android studio. With all these requirements in place, you can move on to the steps involved in the integration process.
-Assuming you have your Android studio up and running, the only thing you have to install is the FAN SDK.
+With your Android studio installed and working properly, you will needto install a Facebook Audience Network (FAN) account sdk.
 
 **FAN (Facebook Audience Network) SDK installation**
-To install, you can download your SDK manually from [here](https://developers.facebook.com/resources/audience-network-sdk-6.8.0.zip) (Not recommended) or follow the steps in [Initial integration steps](#initial-integration-steps) to install it the right way.
+To install, you can download your SDK manually from [here](https://developers.facebook.com/resources/audience-network-sdk-6.8.0.zip) (Not recommended) or follow the steps in [Initial integration steps](#initial-integration-steps) to install it using the recommended method.
```
Comment (Jethro-magaji) [inline content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md:28]: What is the recommended method, is it the steps below? if it is, then specify it by telling the reader else explain the recommended method.

**Review by Jethro-magaji — CHANGES_REQUESTED** [review]
Hi @del-ui 
I have dropped some reviews for you. Please use Grammarly and ProWritingAid to edit your article as there are still some grammatical errors. Else the next revision will require the closure of this PR.
Thanks.

**Comment by hectorkambow** [comment]
> Hi @del-ui I have dropped some reviews for you. Please use Grammarly and ProWritingAid to edit your article as there are still some grammatical errors. Else the next revision will require the closure of this PR. Thanks.

@Jethro-magaji 

Why was this PR not closed as indicated? 

image

**Comment by Jethro-magaji** [comment]
> > Hi @del-ui I have dropped some reviews for you. Please use Grammarly and ProWritingAid to edit your article as there are still some grammatical errors. Else the next revision will require the closure of this PR. Thanks.
> 
> @Jethro-magaji
> 
> Why was this PR not closed as indicated?
> 
> image

He made some the required changes that's why I left it open.

**Comment by Jethro-magaji** [comment]
Hi! @ahmadmardeni1 
I have done my part of the review, please complete the review process by adding a format matter and other necessary tags.
Thanks.

**Comment by WanjaMIKE** [comment]
@ahmadmardeni1, @Kayere, @LinusMuema

**Comment by lalith1403** [comment]
Jumping on this one 🚀

**Comment by lalith1403** [comment]
Hello @Jethro-magaji @del-ui, Can you shorten the URL of the article? It is way too long.

File: content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md [hunk 14]
```diff
@@ -0,0 +1,186 @@
...
+### Demerits of integrating in-app ads into some Applications
+In-App advertisement, as seen earlier, has many advantages. Besides, it is not always Appealing to some users when ads keep popping in all the time.
+
+Here are some demerits in an in-App advertisement;
+
+- **Disturbances from ads pop-up** create a user's negative view towards the application. This may discourage users from using applications known to have in-App advertisements.
+
+- **Difficult to navigate** - Some interested customers may not get to see the advertisement due to the incompatibility of the ads with some phones.
+
+- **Cost incurred on the user** to get the advertisement. To get the advertisement, the user's internet bundles get consumed from an unintended advertisement viewing task.
```
Comment (lalith1403) [inline content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md:176]: This point isn't clear. The user's internet bundle? If you are referring to cellular data, it may be better to reword it.

File: content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md [hunk 15]
```diff
@@ -0,0 +1,186 @@
...
+The key requirements for a successful integration include:
+- [Facebook Audience Network SDK](https://developers.facebook.com/resources/audience-network-sdk-6.7.0.zip)
+- [Facebook Audience Network account](https://business.facebook.com/pub/home/?source=help&business_id=1101812683558213&global_scope_id=1101812683558213)
+- [Android Studio](https://developer.android.com/studio/releases)
+- [Facebook business account](https://business.facebook.com/overview)
+
+You also need to be familiar with the layout and Android studio. With all these requirements in place, you can move on to the steps involved in the integration process.
+With your Android studio installed and working correctly, you will need to install a Facebook Audience Network (FAN) account SDK.
+
+**FAN (Facebook Audience Network) SDK installation**
```
Comment (lalith1403) [inline content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md:55]: If this is a subsection, use H4 heading.

File: content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md [hunk 16]
```diff
@@ -0,0 +1,186 @@
...
+To install, you can download your SDK manually from [here](https://developers.facebook.com/resources/audience-network-sdk-6.8.0.zip) (Not recommended) or follow the steps in [Initial integration steps](#initial-integration-steps) below to install it using the recommended method.
+
+### Initial integration steps
+Before beginning the integration process, you first set up your Facebook Business Manager account for the ads.
+The steps below will guide you in setting up your account.
+
+#### Step 1. **Set-up a  Business Page**
```
Comment (lalith1403) [inline content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md:62]: No need to bold headings

File: content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md [hunk 17]
```diff
@@ -0,0 +1,186 @@
...
+With all that done, run your code and observe the output.
+For successful integration, the following output is obtained.
+
+![output](engineering-education/integrating-facebook-ads-with-your-android-application-in-android-studio
+/output.png)
+
+You can alter the code to and other types of Ads like interstitial Ads and Native Ads in the same manner.
+
+### Why integrate Facebook ads?
+Consumers prefer in-App advertising over display advertising on mobile websites. Smartphone consumers find ads on the mobile web to be compacted and unpleasant. In-App advertisements are resized to fit the screen and seem better, increasing overall engagement.
```
Comment (lalith1403) [inline content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md:163]: Is this true though? @hectorkambow Need your help to review this para.

Comment (lalith1403) [reply content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md:163]: *this section

**Review by lalith1403 — CHANGES_REQUESTED** [review]
Several errors and readability issues exist over a glance. Please rectify these and let me know. The URL needs to be crisp and short. Make the change and make sure to change them in all instances.

**Comment by lalith1403** [comment]
Hello @del-ui @Jethro-magaji. There are several readability issues with this article even after 4 rounds of revisions. I will go ahead and close this indefinitely. Please refer to the docs in the repo to understand the quality of articles we expect.
